ts/features/bonus/bpd/screens/details/components/summary/TextualSummary.tsx

```diff
@@ -0,0 +1,155 @@
+  name: string | undefined;
+};
+
+/**
+ * Display a warning for the current period if transactions < minTransaction and status === "Active"
+ */
+const CurrentPeriodWarning = (props: { period: BpdPeriod }) => (
+  <InfoBox>
+    <Body>
+      {I18n.t(
+        "bonus.bpd.details.components.transactionsCountOverview.currentPeriodKOBody",
+        {
+          transactions: props.period.minTransactionNumber
+        }
+      )}
+    </Body>
+  </InfoBox>
+);
+
+/**
+ * The user doesn't receive the amount (not enough transactions for the closed period)
+ * @param props
+ * @constructor
+ */
+const ClosedPeriodKO = (props: Props) => (
+  <InfoBox>
+    <Body>
+      {I18n.t(
+        "bonus.bpd.details.components.transactionsCountOverview.closedPeriodKOBody",
+        {
+          transactions: props.period.minTransactionNumber,
+          amount: props.amount.totalCashback
+        }
+      )}
+    </Body>
+  </InfoBox>
+);
+
+/**
+ * The user will receive the refund!
+ * @param props
+ * @constructor
+ */
+const ClosedPeriodOK = (props: Props) => (
+  <InfoBox iconName={"io-complete"}>
+    <Body>
+      {I18n.t(
+        "bonus.bpd.details.components.transactionsCountOverview.closedPeriodOKBody",
+        {
+          name: props.name,
+          amount: props.amount.totalCashback
+        }
+      )}
+      {/* If the max amount is reached, inform the user */}
+      {props.amount.totalCashback >= props.period.maxPeriodCashback
+        ? ": " +
+          I18n.t(
+            "bonus.bpd.details.components.transactionsCountOverview.closedPeriodMaxAmount"
+          )
+        : "!"}
+    </Body>
+  </InfoBox>
+);
+
+/**
+ * We await receipt of the latest transactions to consolidate the count
+ * @param props
+ * @constructor
+ */
+const GracePeriod = (props: { period: BpdPeriod }) => (
+  <InfoBox iconName={"io-timer"}>
+    <Body>
+      {I18n.t(
+        "bonus.bpd.details.components.transactionsCountOverview.gracePeriodBody",
+        {
+          date: dateToAccessibilityReadableFormat(props.period.endDate)
+        }
+      )}
+    </Body>
+  </InfoBox>
+);
+
+/**
+ * Inform the user about the start date of the next period
+ * @param props
+ * @constructor
+ */
+const InactivePeriod = (props: { period: BpdPeriod }) => (
+  <InfoBox>
+    <Body>
+      {I18n.t(
+        "bonus.bpd.details.components.transactionsCountOverview.inactivePeriodBody",
+        {
+          date: dateToAccessibilityReadableFormat(props.period.startDate)
+        }
+      )}
+    </Body>
+  </InfoBox>
+);
+
+/**
+ * Choose the textual infobox based on period and amount values
+ * @param props
+ */
+const chooseTextualInfoBox = (props: Props) => {
+  switch (props.period.status) {
+    case "Inactive":
+      return <InactivePeriod period={props.period} />;
+    case "Closed":
+      const today = new Date();
+      const endGracePeriod = new Date();
+      endGracePeriod.setDate(
+        props.period.endDate.getDate() + props.period.gracePeriod
```

Comment:
if `props.period.gracePeriod` express the days, perhaps should it be converted in millisecond?
`props.period.endDate.getDate() + props.period.gracePeriod * 1000 * 60 * 60 * 24`

Reply:
This should be right according to the documentation https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/setDate

Reply:
My fault! I didn't test it and I confused `getDate` with `getTime`